experimental/net/net.go

```diff
@@ -0,0 +1,90 @@
+package net
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+)
+
+// NetConfig is a configuration for TCPListeners.
+type NetConfig interface {
+	ToTCPListeners() ([]*net.TCPListener, error)
+}
+
+// NewNetConfig is a constructor for NetConfig.
+func NewNetConfig(ctx context.Context) (NetConfig, error) {
+	value := ctx.Value(TCPListenersKey{})
+	if value != nil {
+		if tcpListeners, ok := value.(TCPListeners); ok {
+			return &netConfig{tcpListeners: tcpListeners}, nil
+		} else {
+			return nil, fmt.Errorf("invalid type at context key TCPListenersKey, %T", value)
+		}
+	}
+	return &netConfig{}, nil
+}
+
+type netConfig struct {
+	tcpListeners TCPListeners
+}
+
+func (c *netConfig) ToTCPListeners() (listeners []*net.TCPListener, err error) {
```

Comment:
move code like this directly into /runtime.go as this is a config type and it is easier to test when only doing that. For example, `FSConfig` doesn't have logic to open files, so this shouldn't have logic to create the listeners, only to specify them.

On chat what I meant was this basically, this object just holds the listener strings and documents what to put in (like host port rules).

just under where fs stuff happens, you'd do something like this.

```go
	var listeners []*net.TCPListener // normal go net package

	if n, ok := ctx.Value(internalnet.NetConfigKey{}).(*internalnet.NetConfig); ok {
		if listeners, err = f.ToTCPListeners(); err != nil {
			return
		}
	}
```

Then,`listeners` is passed to `internalsys.NewContext` as the last parameter and inserted with tests to make sure that when the context is closed, they are too.